server/routes/prisonerList/pagedPrisonerSearchSummary.ts

```diff
@@ -0,0 +1,219 @@
+import moment from 'moment'
+import type { PrisonerSearchSummary } from 'viewModels'
+
+/**
+ * A class encapsulating an array of [PrisonerSearchSummary] records, exposing them as a paged collection.
+ * The class exposes methods to sort, filter, change page and return the current page of records.
+ *
+ * All fields that represent indexes (`currentPageNumber`, `resultIndexFrom` and `resultIndexTo`) are 1 indexed to make
+ * it easier for view concerns to interact with this class.
+ *
+ * The internal state of this class is mutable. Of particular note is the internal array of [PrisonerSearchSummary] records
+ * which is mutated by the `filter` method. Calling the `filter` method removes elements from the internal array of
+ * [PrisonerSearchSummary] records, and this operation is not reversible.
+ */
+export default class PagedPrisonerSearchSummary {
+  private prisonerSearchSummaries: PrisonerSearchSummary[]
+
+  currentPageNumber: number
+
+  totalPages: number
+
+  pageSize: number
+
+  totalResults: number
+
+  resultIndexFrom: number
+
+  resultIndexTo: number
+
+  /**
+   * Construct a new [PagedPrisonerSearchSummary] instance with the specified array of [PrisonerSearchSummary] records
+   * and page size.
+   */
+  constructor(prisonerSearchSummaries: PrisonerSearchSummary[], pageSize: number) {
+    this.prisonerSearchSummaries = prisonerSearchSummaries
+    this.pageSize = pageSize
+    this.totalPages = Math.ceil(prisonerSearchSummaries.length / pageSize)
+    this.setCurrentPageNumber(1)
+  }
+
+  /**
+   * Sets the current page number.
+   * Attempting to set the current page number to less than 1 will set the current page number to 1.
+   * Attempting to set the current page number to a number greater than the total number of pages will set the current
+   * page number to the last page number.
+   *
+   * This method also updates the `resultIndexFrom` and `resultIndexTo` fields to allow view concerns to be able to render
+   * content such as "Showing records x thru y"
+   */
+  setCurrentPageNumber = (pageNumber: number): PagedPrisonerSearchSummary => {
+    this.currentPageNumber = Math.min(this.totalPages, Math.max(1, pageNumber))
+    this.totalResults = this.prisonerSearchSummaries.length
+    this.resultIndexFrom = Math.min(1 + (this.currentPageNumber - 1) * this.pageSize, this.totalResults)
```

Comment:
Minor - I know that the addition (+ 1) happens after the multiplication because it takes precedence, but what about surrounding it with an extra set of parenthesis? I think it helps improve readability (but like I said, this is a minor comment!):
```
1 + ((this.currentPageNumber - 1) * this.pageSize)
```

Reply:
Interesting you should suggest this - I did exactly this when I first wrote it, and the linter told me off and said the parenthesis were unnecessary and made me remove them! 🤣 🤷‍♂️